### PR TITLE
remove helm deployed backup/restore e2e test case from default test

### PR DESCRIPTION
**Pull request by dragonly**
### What problem does this PR solve?

improve e2e stability

### What is changed and how does it work?

remove helm deployed backup/restore e2e test case from default test.

since it is recommended to use CR to manage backup and restore job, we should remove this and add another e2e test case using CR later

### Code changes

- [x] Has Go code change
- [ ] Has CI related scripts change

### Tests

- [ ] Unit test 
- [ ] E2E test 
- [ ] Manual test 
- [x] No code 

### Side effects

- [ ] Breaking backward compatibility 
- [ ] Other side effects: 

### Related changes

- [x] Need to cherry-pick to the release branch 
- [ ] Need to update the documentation 

### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note
None
```

### CODE REVIEW

File: tests/e2e/tidbcluster/stability.go [hunk 1]
```diff
@@ -1591,5 +1591,27 @@ var _ = ginkgo.Describe("[Stability]", func() {
 			err = cli.PingcapV1alpha1().TidbClusterAutoScalers(tcas.Namespace).Delete(tcas.Name, &metav1.DeleteOptions{})
 			framework.ExpectNoError(err, "failed to delete auto-scaler")
 		})
+
+		// TODO: remove helm deployed backup/restore, use CR instead
```
Comment (DanielZhangQD) [inline tests/e2e/tidbcluster/stability.go:1595]: We can just remove this case and there is case for BR with CR in `stability-br.go`.

Comment (dragonly) [reply tests/e2e/tidbcluster/stability.go:1595]: 👌

File: tests/e2e/tidbcluster/tidbcluster.go [hunk 2]
```diff
@@ -296,27 +296,6 @@ var _ = ginkgo.Describe("TiDBCluster", func() {
...
-		oa.DeployTidbClusterOrDie(&tcCfgFrom)
-		oa.DeployTidbClusterOrDie(&tcCfgTo)
-		oa.CheckTidbClusterStatusOrDie(&tcCfgFrom)
-		oa.CheckTidbClusterStatusOrDie(&tcCfgTo)
-		oa.CheckDisasterToleranceOrDie(&tcCfgFrom)
-		oa.CheckDisasterToleranceOrDie(&tcCfgTo)
-
-		// backup and restore
-		ginkgo.By(fmt.Sprintf("Backup %q and restore into %q", tcCfgFrom.ClusterName, tcCfgTo.ClusterName))
-		oa.BackupRestoreOrDie(&tcCfgFrom, &tcCfgTo)
```
Comment (csuzhangxc) [inline tests/e2e/tidbcluster/tidbcluster.go:317]: do we need to remove this method?

**Review by csuzhangxc — APPROVED** [review]
LGTM

File: tests/e2e/tidbcluster/tidbcluster.go (hunk 2 above)
Comment (dragonly) [reply tests/e2e/tidbcluster/tidbcluster.go:317]: I think we can leave it here for later reference. We can later remove all unneeded test functions.

**Review by DanielZhangQD — APPROVED** [review]
LGTM

**Pull request merged**